src/Model/Action.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Model;
+
+use atk4\core\DIContainerTrait;
+use atk4\core\Exception;
+use atk4\core\InitializerTrait;
+use atk4\core\TrackableTrait;
+use atk4\data\Model;
+
+/**
+ * Implements generic user action. Assigned to a model it can be invoked by a user. Model\Action class contains a
+ * meta information about the action (arguments, permissions, scope, etc) that will help UI/API or add-ons to display
+ * action trigger (button) correctly in an automated way.
+ *
+ * Action must NOT rely on any specific UI implementation.
+ *
+ * @property Model $owner
+ */
+class Action
```

Comment:
> Does actions makes sense for backend only? (or are they user UI/GUI helpers)

I think it may be ok to keep `UserAction` if strictly related with GUI - now it seems so, as it basically add confirmations, title, ... For new users, it is also much more explanatory.

Reply:
Switched to UserAction - it makes sense. Also changes done in atk4/ui